tmail-backend/integration-tests/jmap/jmap-integration-tests-common/src/main/scala/com/linagora/tmail/james/common/LabelSetMethodContract.scala

```diff
@@ -0,0 +1,537 @@
+      .isEqualTo(
+        s"""{
+           |  "sessionState": "${SESSION_STATE.value}",
+           |  "methodResponses": [
+           |    ["Label/set", {
+           |      "accountId": "$ACCOUNT_ID",
+           |      "notCreated": {
+           |        "L13": {
+           |          "type": "invalidArguments",
+           |          "description": "Some server-set properties were specified",
+           |          "properties": ["id", "keyword"]
+           |        }
+           |      }
+           |    }, "c1"]
+           |  ]
+           |}""".stripMargin)
+  }
+
+  @Test
+  def sessionShouldAdvertiseCapability(): Unit = {
+    val sessionJson: String = `given`()
+    .when()
+      .header(ACCEPT.toString, ACCEPT_RFC8621_VERSION_HEADER)
+      .headers(getHeadersWith(BOB_BASIC_AUTH_HEADER))
+      .get("/session")
+    .`then`
+      .statusCode(SC_OK)
+      .contentType(JSON)
+      .extract()
+      .body()
+      .asString()
+
+    assertThatJson(sessionJson)
+      .inPath("capabilities.com:linagora:params:jmap:labels")
+      .isEqualTo("{}")
+  }
+
+  @Test
+  def labelSetCreateShouldSucceedWithDelegatedAccount(server: GuiceJamesServer): Unit = {
+    server.getProbe(classOf[DelegationProbe])
+      .addAuthorizedUser(BOB, ANDRE)
+
+    val request =
+      s"""{
+         |  "using": ["urn:ietf:params:jmap:core", "com:linagora:params:jmap:labels"],
+         |  "methodCalls": [
+         |    ["Label/set", {
+         |      "accountId": "$ACCOUNT_ID",
+         |      "create": {
+         |        "L13": {
+         |          "displayName": "$LABEL_NAME",
+         |          "color": "$LABEL_COLOR"
+         |        }
+         |      }
+         |    }, "c1"]
+         |  ]
+         |}""".stripMargin
+
+    val response = `given`(baseRequestSpecBuilder(server)
+        .setAuth(authScheme(UserCredential(ANDRE, ANDRE_PASSWORD)))
+        .addHeader(ACCEPT.toString, ACCEPT_RFC8621_VERSION_HEADER)
+        .build)
+      .body(request)
+    .when()
+      .post()
+    .`then`
+      .log().ifValidationFails()
+      .statusCode(HttpStatus.SC_OK)
+      .contentType(JSON)
+      .extract()
+      .body()
+      .asString()
+
+    val label: Label = server.getProbe(classOf[JmapGuiceLabelProbe])
+      .listLabels(BOB)
+      .get(0)
+
+    assertThatJson(response)
+      .whenIgnoringPaths("methodResponses[0][1].newState", "methodResponses[0][1].oldState")
+      .isEqualTo(
+        s"""{
+           |  "sessionState": "${SESSION_STATE.value}",
+           |  "methodResponses": [
+           |    ["Label/set", {
+           |      "accountId": "$ACCOUNT_ID",
+           |      "created": {
+           |        "L13": {
+           |          "id": "${label.id.id.value}",
+           |          "keyword": "${label.keyword.flagName}"
+           |        }
+           |      }
+           |    }, "c1"]
+           |  ]
+           |}""".stripMargin)
+  }
```

Comment:
I see the processingContext record clientId,serverId in Method. But do not see "shouldSupportBackreference" here